syntax = "proto3";

package api;

option csharp_namespace = "Nutritracker.DarkArts.Analytics.Api";

message GetCaloriesRequest {
  int32 user_id = 1;
  string start_date = 2;
  string end_date = 3;
}

message GetRecsRequest {
  int32 user_id = 1;
}

message GetCaloriesResponse {
  float total_calories = 1;
}

message GetAllCaloriesResponse {
  repeated float calories = 1;
}

message Recommendation {
  int32 dish_id = 1;
  string dish_name = 2;
  int32 amount = 3;
}

message GetRecsResponse {
  repeated Recommendation recs = 1;
}


service AnalyticService {
  rpc GetCaloriesStatistic(GetCaloriesRequest) returns (GetCaloriesResponse);
  rpc GetAllCaloriesStatistic(GetCaloriesRequest) returns (GetAllCaloriesResponse);
  rpc GetRecommendations(GetRecsRequest) returns (GetRecsResponse);
}
